sql/core/src/main/scala/org/apache/spark/sql/execution/metric/CustomMetrics.scala

```diff
@@ -47,25 +47,21 @@ object CustomMetrics {
 }
 
 /**
- * Built-in `CustomMetric` that sums up metric values.
+ * Built-in `CustomMetric` that sums up metric values. Note that please extend this class
+ * and override `name` and `description` to create your custom metric for real usage.
  */
-class CustomSumMetric extends CustomMetric {
-  override def name(): String = "CustomSumMetric"
-
-  override def description(): String = "Sum up CustomMetric"
+abstract class CustomSumMetric extends CustomMetric {
```

Comment:
This is in the internal package. Shall we move it to the v2 package and rewrite it using java?

Reply:
or we plan to use it for internal sources only?

Reply:
I will create a followup to do it.